build_tools/build_ext.py

```diff
@@ -73,6 +72,17 @@ def _build_cmake(self, build_dir: Path, install_dir: Path) -> None:
         build_command = [_cmake_bin, "--build", build_dir]
         install_command = [_cmake_bin, "--install", build_dir]
 
+        # Check whether parallel build is restricted
+        max_jobs = get_max_jobs_for_parallel_build()
+        if found_ninja():
+            configure_command.append("-GNinja")
+            if max_jobs > 0:
+                build_command.extend(["--parallel", str(max_jobs)])
+        else:
+            build_command.extend(["-j"])
+            if max_jobs > 0:
+                build_command.extend([str(max_jobs)])
```

Comment:
Nit: since we're building through CMake, we can use the same [`--parallel` API](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-build-j) for Ninja and make.
```suggestion
        # Check whether parallel build is restricted
        max_jobs = get_max_jobs_for_parallel_build()
        if found_ninja():
            configure_command.append("-GNinja")
        build_command.append("--parallel")
        if max_jobs > 0:
            build_command.append(str(max_jobs))
```